### PR TITLE
HexChat plugin

### PR DESCRIPTION
**Pull request by Lockyc**
Hey OJ,

I was speaking with Justin, he mentioned you had this plugin and that some people might want a hexchat version (Justin and I will end up using it).
So I ported it, I basically tried to only touch ui_\* functions and substite the irssi api with hexchat. I did notice one bug that i can currently still remember (msg_received doesnt check if aescrypt is active) i fixed it in the hexchat port.

I had to add Try::Tiny to error check the json decoding as it was intermittently crashing hexchat. I probably could have added a few things and rearranged things, but i figured it would be best not to for this pull.

If you accept the pull, a good idea might be to move the crypto functions to a reference file. But I haven't looked into the feasibility of this.

### CODE REVIEW

**Comment by OJ** [comment]
Hey Locky! Cheers for this dude. I'll have a look through the code today. Looks like a great submission.

Catch ya soon mate.

File: hexchat/aescrypt.pl [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+$VERSION = "0.1.0";
+%IRSSI = (
+	authors => 'OJ Reeves',
+	contact => 'oj@buffered.io',
+	name => 'aescrypt',
+	description => 'This script encrypts IRC communications using AES in CBC mode. The script supports per-user and per-channel keys and IVs which can be set via irssi commands. Encrypted content is compresed and Base64 encoded prior to being transmitted.',
+	license => 'Apache 2.0',
+	url => 'https://github.com/OJ/aescript'
+);
+HexChat::register($IRSSI{name}, $VERSION, $IRSSI{description});
```
Comment (OJ) [inline hexchat/aescrypt.pl]: I'd be happy for you to rip out all the IRSSI related stuff completely and just have HexChat related stuff instead mate. These headers also give me too much credit for your own stuff. They should reflect your effort, not mine, and just say "based on.." :)

File: hexchat/aescrypt.pl [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+#     - Payloads sent as JSON, with checksum for payload validation
+#     - Single messages via /aes <message>
+#     - Enable for all messages via /aeson
+#     - Disable for all messages via /aesoff
+#     - Show configuration via /aesshow
+#     - Doesn't screw with bells and other irssi features like the
+#       blowjob.pl script does
+
+use strict;
+#use warnings;
```
Comment (OJ) [inline hexchat/aescrypt.pl]: Feel free to remove commented out stuff mate.

**Comment by OJ** [comment]
Wow, this is really very similar to the irssi version. I didn't realise it'd be that simple! Have you guys tested this to see if it works with one person on HexChat and the other on irssi?

Cheers :)

**Comment by Lockyc** [comment]
Yeah its pretty similar, I just grabbed your existing code and changed the irssi refferences to hexchat refferences. I have done some testing myself with a irssi client and a hexchat client. As far as i can tell it all seems to be working, but i suppose its not extensive testing as i wasnt trying to change any of the core functionality.

**Comment by OJ** [comment]
I have landed this into master. Thanks again dude! I collapsed a bunch of changes into a single changset on the way.

Cheers!

**Comment by Lockyc** [comment]
Cheers! Yeah I didn't know how to do that, I probably should have looked into it.

**Pull request closed without merging**